drivers/flash/flash_stm32_ospi.c

```diff
@@ -1370,11 +1370,17 @@ static int flash_stm32_ospi_read(const struct device *dev, off_t addr,
 	cmd.AddressSize = stm32_ospi_hal_address_size(dev);
 	/* DataSize is set by the read cmd */
 
+	LOG_DBG("OSPI: read %zu data", size);
+
 	/* Configure other parameters */
 	if (dev_cfg->data_rate == OSPI_DTR_TRANSFER) {
 		/* DTR transfer rate (==> Octal mode) */
 		cmd.Instruction = SPI_NOR_OCMD_DTR_RD;
 		cmd.DummyCycles = SPI_NOR_DUMMY_RD_OCTAL_DTR;
+		/* In DTR mode, Nb of bytes to read must be EVEN : one dummy byte is read */
+		if (size % 2) {
+			size = size + 1;
```

Comment:
Won't this cause an overflow if the buffer pointed to by `data` is exactly `size`-bytes long?